sp/src/game/server/ai_senses.cpp

```diff
@@ -749,4 +786,27 @@ void CAI_SensedObjectsManager::AddEntity( CBaseEntity *pEntity )
 	m_SensedObjects.AddToTail( pEntity );
 }
 
+#ifdef MAPBASE
+void CAI_SensedObjectsManager::RemoveEntity( CBaseEntity *pEntity )
+{
+	int i = m_SensedObjects.Find( pEntity );
+	if (i == m_SensedObjects.InvalidIndex())
+		return;
+
+	pEntity->RemoveFlag( FL_OBJECT );
```

Comment:
What does removing this flag from the entity do?

Reply:
It prevents the entity from being sensed anymore. `FL_OBJECT`'s purpose is to mark non-player, non-NPC entities that should be seen by NPC senses.